village/src/components/SmurfForm.js

```diff
@@ -1,25 +1,31 @@
-import React, { Component } from 'react';
+import React, { Component } from "react";
 
 class SmurfForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      name: '',
-      age: '',
-      height: ''
+      name: "",
+      age: "",
+      height: ""
     };
   }
 
-  addSmurf = event => {
+  addNewSmurf = event => {
     event.preventDefault();
     // add code to create the smurf using the api
+    const newSmurf = { ...this.state };
+    this.props.addSmurf(newSmurf);
 
     this.setState({
-      name: '',
-      age: '',
-      height: ''
+      name: "",
+      age: "",
+      height: ""
     });
-  }
+
+    alert("New Smurf has been successfully added");
+    this.props.history.push("/");
+    window.location.reload();
```

Comment:
I wonder why you need to reload the page here since setState is handling the reload or rerender automatically.